docs/source/example_notebooks/functional_api.ipynb

```diff
@@ -0,0 +1,219 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import dowhy as dw\n",
+    "from dowhy.causal_identifier import (\n",
+    "    AutoIdentifier,\n",
+    "    BackdoorAdjustment,\n",
+    "    IDIdentifier,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from dowhy.causal_graph import CausalGraph\n",
+    "import dowhy.datasets\n",
+    "\n",
+    "data = dowhy.datasets.linear_dataset(\n",
+    "    beta=4,\n",
+    "    num_common_causes=2,\n",
+    "    num_instruments=1,\n",
+    "    num_effect_modifiers=1,\n",
+    "    num_samples=500,\n",
+    "    treatment_is_binary=True,\n",
+    "    stddev_treatment_noise=10,\n",
+    "    num_discrete_common_causes=1,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "treatment_name = data[\"treatment_name\"]\n",
+    "outcome_name = data[\"outcome_name\"]\n",
+    "\n",
+    "graph = CausalGraph(\n",
+    "    treatment_name=treatment_name,\n",
+    "    outcome_name=outcome_name,\n",
+    "    graph=data[\"gml_graph\"],\n",
+    "    effect_modifier_names=data[\"effect_modifier_names\"],\n",
+    "    common_cause_names=data[\"common_causes_names\"],\n",
+    "    observed_node_names=data[\"df\"].columns.tolist(),\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Functional API Usage"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# New functional API (AutoIdentifier)\n",
+    "\n",
+    "identified_estimand_auto = dw.identify_effect(\n",
+    "    graph=graph,\n",
+    "    treatment=treatment_name,\n",
+    "    outcome=outcome_name,\n",
+    "    method=AutoIdentifier(estimand_type=dw.EstimandType.NONPARAMETRIC_ATE),\n",
+    ")\n",
```

Comment:
for this API, can you add an example of how a different BackdoorAdjustment can be applied in line 76? It may also be useful to start with an example that just calls "AutoIdentifier()". So my suggestion is to add two calls using the same API. 
1. Default call using AutoIdentifier(). In markdown, we can specify what those defaults are.
2. Second call as fully parameterized AutoIdentifier with both estimand type and backdoor adjustment type.